ilc/client/navigationEvents/setupEvents.js

```diff
@@ -141,21 +134,20 @@ function patchedUpdateState(updateState) {
          * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/History_API}
          * @see {@link https://github.com/mapbox/scroll-restorer/blob/main/index.js#L65}
          */
-        if (typeof url === 'object' && url.pathname) {
-            url = url.pathname + url.search + url.hash;
+        if (typeof url === 'object' && url.href) {
+            url = url.href;
         }
 
-        if (url) {
-            const hooksRes = callNavigationHooks(url);
+        const oldUrl = window.location.href;
+
+        if (oldUrl !== new URL(url, window.location.origin).href) {
+            url = callNavigationHooks(url);
```

Comment:
Make sure that `url` always contains relative url.

Reply:
Updated